drivers/adc/adc_ads1x1x.c

```diff
@@ -138,11 +143,39 @@ struct ads1x1x_data {
 	int16_t *buffer;
 	int16_t *repeat_buffer;
 	struct k_thread thread;
+	k_tid_t	tid;
 	bool differential;
+	struct gpio_callback gpio_cb;
+	struct k_work work;
```

Comment:
I think both of these would also need a `#ifdef CONFIG_ADC_ADS1X1X_TRIGGER`

Reply:
> #ifdef DT_ANY_INST_HAS_PROP_STATUS_OKAY(alert_rdy_gpios)

How would that work? `DT_ANY_INST_HAS_PROP_STATUS_OKAY` relies on DT_DRV_COMPAT, but that is defined at the end.

This driver handles different instances

Reply:
DT_DRV_COMPAT could be moved to the top, it doesn't have to be at the bottom, but I agree that would overcomplicate this PR, and can be done after this is merged